rest-api-spec/src/main/resources/rest-api-spec/api/_internal.prevalidate_node_removal.json

```diff
@@ -0,0 +1,29 @@
+{
+  "_internal.prevalidate_node_removal": {
+    "documentation":{
+      "url": null,
+      "description":"Prevalidates node removal from the cluster"
+    },
+    "stability":"experimental",
+    "visibility":"private",
+    "headers":{
+      "accept": [ "application/json"]
+    },
+    "url":{
+      "paths":[
+        {
+          "path":"/_internal/prevalidate_node_removal/{node}",
+          "methods":[
+            "GET"
```

Comment:
I think POST is slightly better here, since it mostly runs a function (prevalidate), rather than GETs state. We could support both at least.